models/User.js

```diff
@@ -1,57 +1,37 @@
-const mongoose = require("mongoose");
-const uniqueValidator = require("mongoose-unique-validator");
-const crypto = require("crypto");
-const secret = require("../config").secret;
-
-const UserSchema = new mongoose.Schema(
-    {
-        username: {
-            type: String,
-            lowercase: true,
-            unique: true,
-            required: [true, "can't be blank"],
-            match: [/^[a-zA-Z0-9]+$/, "is invalid"],
-            index: true
-        },
-        email: {
-            type: String,
-            lowercase: true,
-            unique: true,
-            required: [true, "can't be blank"],
-            match: [/\S+@\S+\.\S+/, "is invalid"],
-            index: true
-        },
-        bio: String,
-        image: String,
-        favorites: [{ type: mongoose.Schema.Types.ObjectId, ref: "Article" }],
-        following: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
-        hash: String,
-        salt: String
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
+    id: {
+      allowNull: false,
+      primaryKey: true,
+      type: DataTypes.UUID,
+      defaultValue: DataTypes.UUIDV4
     },
-    { timestamps: true }
-);
-
-UserSchema.plugin(uniqueValidator, { message: "is already taken." });
-
-UserSchema.methods.validPassword = function(password) {
-    const hash = crypto
-        .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-        .toString("hex");
-    return this.hash === hash;
-};
-
-UserSchema.methods.setPassword = function(password) {
-    this.salt = crypto.randomBytes(16).toString("hex");
-    this.hash = crypto
-        .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-        .toString("hex");
-};
-
-UserSchema.methods.toAuthJSON = function() {
-    return {
-        username: this.username,
-        email: this.email
-    };
-};
-
-mongoose.model("User", UserSchema);
+    username: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+      validate: {
+        notEmpty: { msg: 'Empty strings not allowed' }
+      }
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true,
+      validate: {
+        isEmail: { msg: 'Enter a Valid Email' },
+      }
+    },
+    bio: {
+        type: DataTypes.TEXT,
+    },
+    image: {
+        type: DataTypes.STRING,
+},
+    hash: {
+      type: DataTypes.STRING,
+    },
+  });
+  return User;
+};
```

Comment:
- Create a `beforeCreate` hook for hashing passwords before saving
- Create an Instance `toAuthJSON` to  mimic what was there in the mongoose model